utils/agent_remote.go

```diff
@@ -361,3 +361,19 @@ func CreateSkipFileOnSegments(oid string, tableName string, c *cluster.Cluster,
 		return fmt.Sprintf("Could not create skip file %s_skip_%s on segments", fpInfo.GetSegmentPipeFilePath(contentID), oid)
 	})
 }
+
+func StartHelperChecker(cl *cluster.Cluster, fpInfo filepath.FilePathInfo, cancel func()) {
+	go func() {
+		for {
+			time.Sleep(5 * time.Second)
+			remoteOutput := cl.GenerateAndExecuteCommand("Checking gpbackup_helper agent failure", cluster.ON_SEGMENTS, func(contentID int) string {
+				helperErrorFileName := fmt.Sprintf("%s_error", fpInfo.GetSegmentPipeFilePath(contentID))
+				return fmt.Sprintf("! ls %s", helperErrorFileName)
```

Comment:
Why not use CheckAgentErrorsOnSegments function?

Reply:
Because `CheckAgentErrorsOnSegments` will remove the error file, which I do not want to do from the checker goroutine. The main goroutine in turn also will call  `CheckAgentErrorsOnSegments` after unblocking by this checker (and will not see the  error file if we use `CheckAgentErrorsOnSegments` in the checker) - I do not want to spoil this logic.

Reply:
if error then fatal
```go
agentErr := utils.CheckAgentErrorsOnSegments(globalCluster, globalFPInfo)
gplog.FatalOnError(agentErr)
```

Reply:
> agentErr := utils.CheckAgentErrorsOnSegments(globalCluster, globalFPInfo)
> gplog.FatalOnError(agentErr)

Just erroring out by `FatalOnError` without canceling the context will leave hanging COPY operations, though gpbackup will exit with an error.

Using of `CheckAgentErrorsOnSegments()` may introduce race conditions (which goroutine will delete the error file first) in case the main goroutine is not blocked and also checking for helper's errors.
Also, if we error out from here, if the main goroutine is not blocked, we would face race conditions - who will detect the error first, and this will cause different error messages to appear. It will make our tests flaky.

Thus, I think it is better to leave it in the current way - the only checker goroutine's responsibilities will be to probe the helper's error file (without deleting it) and to cancel the query's context.
And leave all error handling as the main goroutine responsibility.